### PR TITLE
feat: #1264 config prod bcsc

### PR DESCRIPTION
**Pull request by MCatherine1994**
refs: #1264

- Config PROD BCSC issuer urls through terraform
- Remove the gender from attribute mapping
- Remove gender from the read/write permission maximum list
- Config AWS DEV to use PROD token url and user info url for BCSC identity provider

### CODE REVIEW

File: infrastructure/server/oidc_idp_bcsc.tf [hunk 1]
```diff
@@ -16,25 +19,24 @@ resource "aws_cognito_identity_provider" "dev_bcsc_oidc_provider" {
     client_secret             = var.dev_oidc_bcsc_idp_client_secret
     oidc_issuer               = var.dev_bcsc_oidc_idp_issuer
     attributes_request_method = "GET"
-    authorize_url = "https://idtest.gov.bc.ca/login/oidc/authorize"
-    token_url = "${aws_api_gateway_deployment.fam_api_gateway_deployment.invoke_url}/bcsc/token/dev"
-    attributes_url = "${var.use_override_proxy_endpoints ?  var.dev_override_bcsc_userinfo_proxy_endpoint : local.dev_local_bcsc_userinfo_proxy_endpoint}"
-
-    jwks_uri = "https://idtest.gov.bc.ca/oauth2/jwk.json"
+    authorize_url             = "https://idtest.gov.bc.ca/login/oidc/authorize"
+    token_url                 = "${var.use_override_proxy_endpoints ? var.dev_override_bcsc_token_proxy_endpoint : local.dev_local_bcsc_token_proxy_endpoint}"
```
Comment (MCatherine1994) [inline infrastructure/server/oidc_idp_bcsc.tf:23]: so we have a `use_override_proxy_endpoints` variable, which will only be used in AWS DEV environment, where we can set the token_url and attribute_url (userinfo url) to use the production api url 
image

File: infrastructure/server/oidc_idp_bcsc.tf [hunk 2]
```diff
@@ -84,26 +84,24 @@ resource "aws_cognito_identity_provider" "prod_bcsc_oidc_provider" {
     client_secret             = var.prod_oidc_bcsc_idp_client_secret
```
Comment (MCatherine1994) [inline infrastructure/server/oidc_idp_bcsc.tf:84]: the secret is getting from Github secret, I already updated it

File: terraform/dev/terragrunt.hcl [hunk 3]
```diff
@@ -43,9 +43,12 @@ generate "dev_tfvars" {
   fam_console_idp_name_bceid = "TEST-BCEIDBUSINESS"
   forest_client_api_base_url = "https://nr-forest-client-api-test.api.gov.bc.ca"
   use_override_proxy_endpoints = true
-  dev_override_bcsc_userinfo_proxy_endpoint = "https://c727z9v3cc.execute-api.ca-central-1.amazonaws.com/v1/bcsc/userinfo/dev"
-  test_override_bcsc_userinfo_proxy_endpoint = "https://c727z9v3cc.execute-api.ca-central-1.amazonaws.com/v1/bcsc/userinfo/test"
-  prod_override_bcsc_userinfo_proxy_endpoint = "https://c727z9v3cc.execute-api.ca-central-1.amazonaws.com/v1/bcsc/userinfo/prod"
+  dev_override_bcsc_userinfo_proxy_endpoint = "https://6mud7781pe.execute-api.ca-central-1.amazonaws.com/v1/bcsc/userinfo/dev"
```
Comment (MCatherine1994) [inline terraform/dev/terragrunt.hcl:46]: use the AWS PROD (PROD API Gateway) token url and user info url to config dev/test/prod BCSC in AWS DEV, so we can use AWS DEV cognito to test the PROD-BCSC login

File: terraform/tools/terragrunt.hcl [hunk 4]
```diff
@@ -43,10 +43,7 @@ generate "tools_tfvars" {
   fam_console_idp_name = "TEST-IDIR"
   fam_console_idp_name_bceid = "TEST-BCEIDBUSINESS"
   forest_client_api_base_url = "https://nr-forest-client-api-test.api.gov.bc.ca"
-  use_override_proxy_endpoints = true
-  dev_override_bcsc_userinfo_proxy_endpoint = "https://c727z9v3cc.execute-api.ca-central-1.amazonaws.com/v1/bcsc/userinfo/dev"
-  test_override_bcsc_userinfo_proxy_endpoint = "https://c727z9v3cc.execute-api.ca-central-1.amazonaws.com/v1/bcsc/userinfo/test"
-  prod_override_bcsc_userinfo_proxy_endpoint = "https://c727z9v3cc.execute-api.ca-central-1.amazonaws.com/v1/bcsc/userinfo/prod"
+  use_override_proxy_endpoints = false
```
Comment (MCatherine1994) [inline terraform/tools/terragrunt.hcl:46]: we will not use the tools environment for bcsc login anyway, no need to rewrite it

File: infrastructure/server/variables_provided.tf [hunk 5]
```diff
@@ -129,7 +129,7 @@ variable "test_oidc_bcsc_idp_client_id" {
 
 variable "prod_oidc_bcsc_idp_client_id" {
   type = string
-  default = "not.yet.implemented"
+  default = "ca.bc.gov.flnr.fam.prod"
```
[the remark this answers is not in the document]
Comment (MCatherine1994) [reply infrastructure/server/variables_provided.tf:132]: Hi Ian, good catch, I think we do see a 401 for prod, test returns 200, let me fix this, maybe that's the problem
Screen Shot 2024-04-05 at 2 25 26 PM
Screen Shot 2024-04-05 at 2 27 45 PM

[the remark this answers is not in the document]
Comment (MCatherine1994) [reply infrastructure/server/variables_provided.tf:132]: found we use it in Auth Lambda file as well, updated there as well

**Comment by MCatherine1994** [comment]
Hi Ian, I think the `use_override_proxy_endpoints` is not related to the flyway, we only use it here when set up token and user info urls

File: terraform/dev/terragrunt.hcl (hunk 3 above)
Comment (MCatherine1994) [reply terraform/dev/terragrunt.hcl:46]: I think the use_override_proxy_endpoints is not related to the flyway, we only use it here when set up token and user info urls

File: infrastructure/server/variables_provided.tf (hunk 5 above)
[the remark this answers is not in the document]
Comment (ianliuwk1019) [reply infrastructure/server/variables_provided.tf:132]: Thank you for updating them. Let see if that solve the problem.

**Review by ianliuwk1019 — APPROVED** [review]

**Pull request merged**